test/ngRoute/directive/ngViewSpec.js

```diff
@@ -913,9 +912,11 @@ describe('ngView animations', function() {
 
       $location.path('/foo');
       $rootScope.$digest();
-      expect($animate.queue.shift().event).toBe('enter');
-      $animate.triggerCallbacks();
 
+      $animate.flush();
+      $rootScope.$digest();
```

Comment:
So beforehand what happened was:

1. `triggerCallbacks()` issued a `$$rAF.flush()` internally when called.
2. `triggerCallbacks()` also calls `$timeout.flush(0)` which calls a digest somewhere down the line.

But with the changes to use `$animate.flush()` we end up having to do this:

1. `$animate.flush()` clears out the RAF for the runner.
2. `$rootScope.$digest()` allows for the resolved promise to flush (which can only be done after the RAF has flushed).

So we need to call RAF + digest always.